vercel_blob/blob_store.py

```
@@ -434,8 +441,24 @@ def download_file(url: str, path: str = '', options: dict = None, timeout: int =
         except FileNotFoundError as e:
             if _DEBUG:
                 print(f"An error occurred. Please try again. Error: {e}")
-            raise Exception("The directory must exist before downloading the file. Please create the directory and try again.")
+            raise BlobFileError("The directory must exist before downloading the file. Please create the directory and try again.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)